docs/syntax_and_semantics/assignment.md

```diff
@@ -171,6 +159,97 @@ objects[1] = temp1
 objects[2] = temp2
 ```
 
+## One-to-many assignment
+
+If the right-hand side contains just one expression, the type is indexed for each variable on the left-hand side like so:
+
+```crystal
+name, age, source = "Crystal, 123, GitHub".split(", ")
+
+# The above is the same as this:
+temp = "Crystal, 123, GitHub".split(", ")
+name = temp[0]
+age = temp[1]
+source = temp[2]
+```
+
+Additionally, if the [`strict_multi_assign` flag](compile_time_flags.md) is provided, the number of elements must match the number of targets, and the right-hand side must be an [`Indexable`](https://crystal-lang.org/api/Indexable.html):
+
+```crystal
+name, age, source = "Crystal, 123, GitHub".split(", ")
+
+# The above is the same as this:
+temp = "Crystal, 123, GitHub".split(", ")
+if temp.size != 3 # number of targets
+  raise IndexError.new("Multiple assignment count mismatch")
+end
+name = temp[0]
+age = temp[1]
+source = temp[2]
+
+a, b = {0 => "x", 1 => "y"} # Error: right-hand side of one-to-many assignment must be an Indexable, not Hash(Int32, String)
+```
+
+## Splat assignment
```

Comment:
```suggestion
### Splat assignment
```

Reply:
I don't think this change is correct. Splats also work if there are several targets on the right-hand side, as in `a, *_, b = 1, 2, 3, 4`. The previous section is for when you have just one expression on the right.

Reply:
Ah, my comment is locally correct but globally wrong :-) We need to increase depth level of he previous one too.